proof.go

```diff
@@ -200,10 +167,47 @@ func (proof Proof) verifyLeafHashes(nth *Hasher, verifyCompleteness bool, nID na
 		}
 	}
 
-	// add remaining proof hashes after the last range ends
-	consumeUntil(math.MaxUint64)
+	var computeRoot func(start, end int) []byte
+	computeRoot = func(start, end int) []byte {
```

Comment:
[optional]

if I'm not mistaken, I think we could use the [one tendermint uses](https://github.com/tendermint/tendermint/blob/5013bc3f4a6d64dcc2bf02ccc002ebc9881c62e4/crypto/merkle/tree.go#L7-L21). If that's the case, I'd be okay with copy/pasting that code here and using it. This is an optional idea tho, this code appears to work fine.